docs/how-to/hip_runtime_api/initialization.rst

```diff
@@ -0,0 +1,102 @@
+.. meta::
+   :description: Initialization.
+   :keywords: AMD, ROCm, HIP, initialization
+
+********************************************************************************
+Initialization
+********************************************************************************
+
+The initialization involves setting up the environment and resources needed for
+using GPUs. The following steps are covered with the initialization:
+
+- Setting up the HIP runtime
+
+  This includes loading necessary libraries and setting up internal data
+  structures.
+
+- Querying and setting GPUs
+
+  Identifying and querying the available GPU devices on the system.
+
+- Setting up contexts
+
+  Creating contexts for each GPU device, which are essential for managing
+  resources and executing kernels. For further details, check the :ref:`context
+  section <context_driver_api>`.
+
+Initialize the HIP runtime
+================================================================================
+
+The HIP runtime is initialized automatically when the first HIP API call is
+made. However, you can explicitly initialize it using :cpp:func:`hipInit`,
+to be able to control the timing of the initialization. The manual
+initialization can be useful to ensure that the GPU is initialized and
+ready, or to isolate GPU initialization time from other parts of
+your program.
+
```

Comment:
We can add a little more detail on what is being done during HIP Initialization, such as:
- Reading the environment variables set during init, and setting up the active or visible devices etc.
- Initializes the compiler as well as HSA runtime and checks any errors due to lack of resources or no active devices etc.
- Also, sets up some internal buffers which will be used for memory copies or cooperative launches.

Reply:
Updated the description at the previous section.